OrbitQt/topdownwidget.cpp

```diff
@@ -0,0 +1,151 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "topdownwidget.h"
+
+#include <QMenu>
+
+#include "TopDownViewItemModel.h"
+
+void TopDownWidget::SetTopDownView(std::unique_ptr<TopDownView> top_down_view) {
+  auto* model =
+      new TopDownViewItemModel{std::move(top_down_view), ui_->topDownTreeView};
+  auto* proxy_model = new QSortFilterProxyModel{ui_->topDownTreeView};
+  proxy_model->setSourceModel(model);
+  proxy_model->setSortRole(Qt::EditRole);
+  ui_->topDownTreeView->setModel(proxy_model);
+  ui_->topDownTreeView->sortByColumn(TopDownViewItemModel::kInclusive,
+                                     Qt::DescendingOrder);
+  ui_->topDownTreeView->header()->resizeSections(QHeaderView::ResizeToContents);
+}
+
+const std::string TopDownWidget::kActionExpandRecursively =
+    "&Expand recursively";
+const std::string TopDownWidget::kActionCollapseRecursively =
+    "&Collapse recursively";
+const std::string TopDownWidget::kActionCollapseChildrenRecursively =
+    "Collapse children recursively";
+const std::string TopDownWidget::kActionExpandAll = "Expand all";
+const std::string TopDownWidget::kActionCollapseAll = "Collapse all";
+
+static void ExpandRecursively(QTreeView* tree_view, const QModelIndex& index) {
```

Comment:
Maybe I missed it, but is there a reason not to use QTreeView::expandRecursively?

Reply:
You caught me off guard for a moment, as I had tried and I didn't remember, but the reason is that it should be there from Qt 5.13 but on my Linux machine I can only get Qt up to 5.12.5... So either I leave it like this or I'll need help.

Reply:
Also there's no `QTreeView::collapseRecursively` so might as well implement the expand version ourselves as well.